datalad/resources/procedures/cfg_noannex.py

```diff
@@ -0,0 +1,41 @@
+"""Procedure to uninit Git-annex (if initialized), and create/save .noannex file to prevent annex initialization
+
+If there are git-annex'ed files already, git annex uninit and this procedure will fail.
+
+"""
+
+from datalad.distribution.dataset import require_dataset
+from datalad.support.annexrepo import AnnexRepo
+from datalad import lgr
+
+
+def no_annex(ds):
+    ds = require_dataset(
+        ds,
+        check_installed=True,
+        purpose='configuration')
+
+    if isinstance(ds.repo, AnnexRepo):
+        repo = ds.repo
+        # TODO: if procedures can have options -- add --force handling/passing
+        #
+        # annex uninit unlocks files for which there is content (nice) but just proceeds
+        # and leaves broken symlinks for files without content.  For the current purpose
+        # of this procedure we just prevent "uninit" of any annex with some files already
+        # annexed.
+        if any(repo.call_annex_items_(['whereis', '--all'])):
+            raise RuntimeError("Annex has some annexed files, unsafe")
```

Comment:
[ just a non-actionable thought while reading ]

The particular output isn't important, so `whereis`s pretty verbose output could be reduced to something like `"--format=.\n"`.  This isn't supported by datalad's minimum git-annex version and would only speedup the error case (and even then, only when there are many files), though.

Reply:
and in principle should not have notable effect here since hopefully annex starts producing records as soon as it has any and our `call_annex_items_` generator yielding them, so `any()` would just stop iterating as soon as it gets at least one

Reply:
> since hopefully annex starts producing records as soon as it has any and our `call_annex_items_` generator yielding them

Perhaps your "in principle" meant you realize this is the case and are looking forward to some future enhancement, but fwiw `call_annex_items_` operates on the entire output of the command.

Reply:
oh...  ["как обманчива природа" сказал ёжик слезая с кактуса](https://translate.google.com/?sl=auto&tl=en&text=%22%D0%BA%D0%B0%D0%BA%20%D0%BE%D0%B1%D0%BC%D0%B0%D0%BD%D1%87%D0%B8%D0%B2%D0%B0%20%D0%BF%D1%80%D0%B8%D1%80%D0%BE%D0%B4%D0%B0%22%20%D1%81%D0%BA%D0%B0%D0%B7%D0%B0%D0%BB%20%D1%91%D0%B6%D0%B8%D0%BA%20%D1%81%D0%BB%D0%B5%D0%B7%D0%B0%D1%8F%20%D1%81%20%D0%BA%D0%B0%D0%BA%D1%82%D1%83%D1%81%D0%B0&op=translate)